In prepareValue, guard debug check against null property access

mk-pmb opened the pull request:
Fixes #748 

It probably just masks the real problem, whatever that might be. For now this makes nodemjs work again, you can still add better tracing later.

Considering how often you use `Loader.state.package.default.options.debug`, it might be worth a guarding helper function.

jdalton commented:
Thank you @mk-pmb!

Yes, `Loader.state.package.default` should be set by then. Are you using `esm` through `node -r esm` or through `require("esm")(module)`?

mk-pmb commented:
the 2nd, [stdEsm = require('esm')](https://github.com/mk-pmb/esmod-pmb-js/blob/master/reexport.js#L4) … [stdEsm(bridgeModule, …)](https://github.com/mk-pmb/esmod-pmb-js/blob/master/reexport.js#L90)

jdalton commented:
Great! I think I've spotted the issue. I'll prepare build of `esm/esm/loader.js` for you to swap out.

jdalton commented:
Okay! You can either build from our master branch or use [loader.js.zip](https://github.com/standard-things/esm/files/2971403/loader.js.zip).

mk-pmb commented:
Looks like no. :-(
I switched to 3.2.12 and replaced `/home/ci-test/lib/node_modules/esm/esm/loader.js` with your new one (SHA-1 099c30ebf88f4506bd7da17eafa6bea7f6d414c3): still fails with `Cannot read property 'options' of null`.
Then I built from master (c07d4ad606477c067ddd4fca04d1218651844453), same result.

jdalton commented:
Hmm OK. I'll add the guard then.

jdalton commented:
Can you try from our master branch again (you'll have to force update since I force pushed)

mk-pmb commented:
Works with current master. :+1:

jdalton commented:
`esm` [v3.2.18](https://github.com/standard-things/esm/releases/tag/3.2.18) is released :tada:
